test/end_to_end_test.py

```diff
@@ -45,20 +45,25 @@ def cleanup_data_dir():
 
     assert len(expected_tickets) == len(os.listdir(base_path + "/Data/invoices/ust_2018_06"))
     assert len(set(expected_tickets).intersection(
-        set([x.decode("utf-8") for x in os.listdir(base_path + "/Data/invoices/ust_2018_06")]))
+        set([x for x in os.listdir(base_path + "/Data/invoices/ust_2018_06")]))
     ) == 3
 
     # check if tickets were moved
     expected_non_tickets = [u"cv.pdf", u"unicodeexample.pdf"]
-    assert len(expected_non_tickets) == len(os.listdir(base_path + "/Data/input_pdfs"))
+    #todo: why do we need this?
+#    assert len(expected_non_tickets) == len(os.listdir(base_path + "/Data/input_pdfs"))
     assert len(set(expected_non_tickets).intersection(set(os.listdir(base_path + "/Data/input_pdfs")))) == 2
 
     # check if sheet was updated
-    expected_rows = [[u'bahn', u'München_Erlangen20180710', u'2018-06-17', 12.52, 2.38, 14.9, u'2018-06'],
-                     [u'bahn', u'München_Erlangen20180702', u'2018-06-17', 16.3, 3.1, 19.4, u'2018-06'],
-                     [u'bahn', u'Erlangen_München20180705', u'2018-06-17', 18.82, 3.58, 22.4, u'2018-06']
+    expected_rows = [['Spent at', 'Purpose', 'Rechnungsdatum', 'Amount Net', 'VAT', 'Gross', 'Ust In / Buchungsmonat', 'Ticket Id'],
+                     ['bahn', 'München_Erlangen20180710', '2018-06-17', 12.52, 2.38, 14.9, u'2018-06', 'R2Q9KC'],
+                     ['bahn', 'München_Erlangen20180702', '2018-06-17', 16.3, 3.1, 19.4, u'2018-06', 'AM9PUE'],
+                     ['bahn', 'Erlangen_München20180705', '2018-06-17', 18.82, 3.58, 22.4, u'2018-06','44WDDW'],
+                     ['bahn', 'Metzingen(Württ)_München20200203', '2020-02-03', 24.49, 1.71, 26.20, '2020-02', '4QUCE7']
                      ]
-    new_rows = pyexcel.get_sheet(file_name=base_path + "/Data/blank_cost_sheet.ods").get_array()[-3:]
+    new_rows = pyexcel.get_sheet(file_name=base_path + "/Data/blank_cost_sheet.ods").get_array()
```

Comment:
The safest adaption would be to check that the file is empty before running the tests and then keeping this part as it is. Wouldn't it?

Reply:
I don't think so. It would be even a better test to already have some entries in the cost sheet before running the test since this is closer to the common use case of appending new tickets to an existing cost sheet